src/cell.rs

```diff
@@ -0,0 +1,131 @@
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum CellError {
+    AliveNeighborOverflow,
+    AliveNeighborUnderflow,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum CellState {
+    Alive,
+    Dead,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Cell {
+    pub state: CellState,
+    pub alive_neighbours: i32,
```

Comment:
Do you really need to store this?  I think it makes the coding kinda complex with results and all.